src/dg/weak_dg.cpp

```diff
@@ -626,6 +628,10 @@ void DGWeak<dim,nstate,real,MeshType>::assemble_volume_term_explicit(
         // Evaluate physical convective flux and source term
         conv_phys_flux_at_q[iquad] = DGBaseState<dim,nstate,real,MeshType>::pde_physics_double->convective_flux (soln_at_q[iquad]);
         diss_phys_flux_at_q[iquad] = DGBaseState<dim,nstate,real,MeshType>::pde_physics_double->dissipative_flux (soln_at_q[iquad], soln_grad_at_q[iquad], current_cell_index);
+        //adding physical source
+        const dealii::Point<dim,real> points = fe_values_vol.quadrature_point(iquad);
+        physical_source_at_q[iquad] = DGBaseState<dim,nstate,real,MeshType>::pde_physics_double->physical_source_term (points,soln_at_q[iquad], soln_grad_at_q[iquad], current_cell_index);
```

Comment:
Have you thought of introducing `has_nonzero_physical_source` in the PhysicsBase class? This way we would avoid computing `physical_source_term` when its zero.

Reply:
fixed.
I add has_nonzero_physical_source, which is 'false' except for the objects of Physics_Model holding RANS.
In weak/strong DG the physical source will be calculated when has_nonzero_physical_source is 'true'.
Thanks for the advice.